src/network/replication/transform.rs

```diff
@@ -0,0 +1,77 @@
+use bevy::{prelude::*, ptr::Ptr};
+use bevy_replicon::{bincode, prelude::*, replicon_core::replication_rules};
+use serde::{Deserialize, Serialize};
+use std::io::Cursor;
+
+use crate::network;
+
+pub struct TransformPlugin;
+
+impl Plugin for TransformPlugin {
+    fn build(&self, app: &mut App) {
+        app.replicate_with::<Transform>(
+            serialize_transform,
+            deserialize_transform,
+            replication_rules::remove_component::<Transform>,
+        )
+        .add_systems(PreUpdate, apply_synced_transform);
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, Component)]
+pub struct SyncedTransform {
+    translation: Vec3,
+    rotation: Quat,
+    scale: Vec3,
+}
+
+impl From<Transform> for SyncedTransform {
+    fn from(value: Transform) -> Self {
+        Self {
+            translation: value.translation,
+            rotation: value.rotation,
+            scale: value.scale,
+        }
+    }
+}
+
+impl From<SyncedTransform> for Transform {
+    fn from(value: SyncedTransform) -> Self {
+        Self {
+            translation: value.translation,
+            rotation: value.rotation,
+            scale: value.scale,
+        }
+    }
+}
+
+fn apply_synced_transform(mut query: Query<(&mut Transform, &SyncedTransform)>, time: Res<Time>) {
+    let t = time.delta_seconds() * network::MAX_TICK_RATE as f32;
+
+    for (mut transform, synced_transform) in query.iter_mut() {
+        transform.translation = transform.translation.lerp(synced_transform.translation, t);
+        transform.rotation = transform.rotation.lerp(synced_transform.rotation, t);
+        transform.scale = synced_transform.scale;
+    }
+}
+
+fn serialize_transform(component: Ptr, cursor: &mut Cursor<Vec<u8>>) -> bincode::Result<()> {
+    // SAFETY: Function called for registered `ComponentId`.
+    let transform: &Transform = unsafe { component.deref() };
```

Comment:
This seems really sus, is this how they do it in replicon's examples?

Besides, why use `replicate_with` instead of `replicate`? It doesn't seem like these replicating functions do anything magical.

Reply:
This line is from replicon's docs.

Any it uses `replicate_with`, because `Transform` do not derrives serialization and deserialization (this can be active), but still we must do magic with lerping.